### DISCUSSION TITLE
Localization handeling

### DISCUSSION THREAD

**Discussion by haringsrob** [area17/twill#1169] [Q&A]
First of all, twill is awesome! It is a great cms builder and 100% in line with what I would build.

I am building a test site, and experimenting a lot currently. However I am running into an issue.

In the function `getLanguagesForVueStore` there is a call to `config('app.locale')`.

However if you have a setup where you only allow one language for pages: `nl` and you have some other call that does a `setLocale('en')` twill basically breaks.

In forms etc, en is populated as the field language, while that language is not enabled.

I want to propose a fix (I can make the pr) that in `getLanguagesForVueStore` we check if `config('app.locale')` is available in `config('translatable.locales')` if not, fall back to the first language in `config('translatable.locales')`.

Any ideas/views on this one?

(I know its an edge case, and I spend quite a bit of time debugging this)

**Comment by pboivin** [comment]
Hi @haringsrob,
At a glance, this looks like a good solution! Don't hesitate to send a PR :)
I'm curious to know more about your setup — where does the `setLocale('en')` call come from in this context?

**Reply by haringsrob** [reply]
It's not really a use case, more like an edge case.

I am currently using (for the front-end) https://github.com/mcamara/laravel-localization,
So in my routes there is `'prefix' => LaravelLocalization::setLocale()` which sets the locale when preparing the routes.

I have (had) setup Laravel localisation to have English and dutch, while only enabled dutch for Twill which led to this edge case.

I don't know if there are really valid use cases for this, once I figured it out it was easy for me to understand, but it was quite the path to debug.

The proposed solution just is a bit more failsafe in case of exotic config.

I will prepare a pr.

**Reply by pboivin** [reply]
Makes sense. So, your final site will be Dutch & English but for now, you only want to allow admins to enter Dutch content?

**Reply by haringsrob** [reply]
Yes exactly :).
